src/modules/order/order.service.js

```
@@ -1,80 +1,480 @@
+  logger.debug(`getAllOrders: Starting to fetch orders for user ${user?.id || 'unknown'} with role ${user?.role || 'unknown'}`);
+  
+  // Check if user object is valid
+  if (!user || (!user.id && !user._id)) {
+    logger.error('getAllOrders: Invalid user object provided');
+    throw new Error("Invalid user information provided");
+  }
+  
   const allowedStatuses = ["pending", "processing", "shipped", "delivered", "cancelled"];
-  const query = allowedStatuses.includes(status) ? { status: { $eq: status } } : {};
-  return await OrderModel.find(query);
+  let query = {};
```

Biome (1.9.4)

[error] This let declares a variable that is only assigned once.

'query' is never reassigned.

Safe fix: Use const instead.


(lint/style/useConst)